WordPress/src/main/java/org/xmlrpc/android/XMLRPCUtils.java

```diff
@@ -289,6 +292,14 @@ private static String verifyXmlrpcUrl(final String siteUrl, final String httpUse
                     // Endpoint found and works fine.
                     return url;
                 }
+            } catch (XMLRPCUtilsException e) {
+                if (e.kind == XMLRPCUtilsException.Kind.ERRONEOUS_SSL_CERTIFICATE ||
+                    e.kind == XMLRPCUtilsException.Kind.HTTP_AUTH_REQUIRED ||
+                    e.kind == XMLRPCUtilsException.Kind.MISSING_XMLRPC_METHOD) {
```

Comment:
99.9% of cases a site has one XML-RPC Endpoint only. When the discovery process has already found it, but some methods are missing in the XML-RPC response of system.listMethods, we can stop the discovery process immediately.